include/glow/Backends/ExecutionContext.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_EXECUTIONCONTEXT_H
+#define GLOW_BACKENDS_EXECUTIONCONTEXT_H
+
+#include "glow/Backends/TraceEvents.h"
+#include "glow/Graph/PlaceholderBindings.h"
+
+#include "llvm/ADT/STLExtras.h"
+
+namespace glow {
+
+/// Sub-classed per backend, this holds Device specific per-function information
+/// if that is necessary on that particular backend.
+class DeviceBindings {
```

Comment:
I think this will need the same solution we used for DeviceConfig, backendKind as a private enum set at construction so the CompiledFunction can check type before casting. (Or the compiledFunction can just trust it is always passed the right type.)

Reply:
Yeah, I think it makes sense to add BackendKind to the base class.